### PR TITLE
OS-8354 piadm should only run in the global zone, and mostly as root

### PR DESCRIPTION
**Pull request by danmcd**
Testing notes in ticket.

### CODE REVIEW

File: src/piadm.sh [hunk 1]
```diff
@@ -41,6 +41,9 @@ corrupt() {
 	exit 3
 }
 
+# Don't run in the global zone.
+[[ "$(zonename)" == "global" ]] || err "Cannot run piadm in the global zone"
```
Comment (wiedi) [inline src/piadm.sh:45]: Should this be the other way around?
```suggestion
# Only run in the global zone.
[[ "$(zonename)" == "global" ]] || err "Cannot run piadm in non-global zone"
```

Comment (danmcd) [reply src/piadm.sh]: My bad.  Fixed.

**Comment by danmcd** [comment]
Some initial testing:
```
[root@smartos-bios ~]# /tmp/piadm.sh -v list
Creating /var/piadm/piadm.conf
Version 1 of /var/piadm/piadm.conf
The following file contents have been configured:

PIADM_CONFIG_VERSION=1
DEFAULT_URL_PREFIX=https://us-east.manta.joyent.com/Joyent_Dev/public/SmartOS/

PI STAMP               BOOTABLE FILESYSTEM            BOOT IMAGE NOW  NEXT 
20220113T011608Z       zones/boot                     available  no   no  
20220118T183559Z       zones/boot                     available  no   no  
20220121T215246Z       zones/boot                     available  no   no  
20220210T011229Z       zones/boot                     next       yes  yes 
[root@smartos-bios ~]# !vi
vi /var/piadm/piadm.conf 
[root@smartos-bios ~]# /tmp/piadm.sh -v list
Version 1 of /var/piadm/piadm.conf
The following file contents have been configured:

PIADM_CONFIG_VERSION=1
DEFAULT_URL_PREFIX=https://us-east.manta.smartos.org/Joyent_Dev/public/SmartOS/

DEFAULT_URL_PREFIX was https://us-east.manta.joyent.com/Joyent_Dev/public/SmartOS/ ,
but now is https://us-east.manta.smartos.org/Joyent_Dev/public/SmartOS/
PI STAMP               BOOTABLE FILESYSTEM            BOOT IMAGE NOW  NEXT 
20220113T011608Z       zones/boot                     available  no   no  
20220118T183559Z       zones/boot                     available  no   no  
20220121T215246Z       zones/boot                     available  no   no  
20220210T011229Z       zones/boot                     next       yes  yes 
[root@smartos-bios ~]# 
```

**Comment by danmcd** [comment]
Sorry for the wrong comment earlier.

**Review by wiedi — APPROVED** [review]

**Review by bahamat — APPROVED** [review]

**Pull request merged**